http-clients/netty-nio-client/src/test/java/software/amazon/awssdk/http/nio/netty/fault/ServerConnectivityErrorMessageTest.java

```diff
@@ -0,0 +1,356 @@
+import software.amazon.awssdk.http.SdkHttpMethod;
+import software.amazon.awssdk.http.SdkHttpResponse;
+import software.amazon.awssdk.http.async.AsyncExecuteRequest;
+import software.amazon.awssdk.http.async.SdkAsyncHttpClient;
+import software.amazon.awssdk.http.async.SdkAsyncHttpResponseHandler;
+import software.amazon.awssdk.http.nio.netty.NettyNioAsyncHttpClient;
+import software.amazon.awssdk.http.nio.netty.SdkEventLoopGroup;
+import software.amazon.awssdk.utils.AttributeMap;
+import software.amazon.awssdk.utils.Logger;
+
+@RunWith(Parameterized.class)
+public class ServerConnectivityErrorMessageTest {
+    private static final Logger LOGGER = Logger.loggerFor(ServerConnectivityErrorMessageTest.class);
+    private SdkAsyncHttpClient netty;
+    private Server server;
+
+    @Parameterized.Parameter
+    public TestCase testCase;
+
+    @Parameterized.Parameters(name = "{0}")
+    public static Collection<TestCase> testCases() {
+        return Arrays.asList(new TestCase(CloseTime.DURING_INIT, "The connection was closed during the request."),
+                             new TestCase(CloseTime.BEFORE_SSL_HANDSHAKE, "The connection was closed during the request."),
+                             new TestCase(CloseTime.DURING_SSL_HANDSHAKE, "The connection was closed during the request."),
+                             new TestCase(CloseTime.BEFORE_REQUEST_PAYLOAD, "The connection was closed during the request."),
+                             new TestCase(CloseTime.DURING_REQUEST_PAYLOAD, "The connection was closed during the request."),
+                             new TestCase(CloseTime.BEFORE_RESPONSE_HEADERS, "The connection was closed during the request."),
+                             new TestCase(CloseTime.BEFORE_RESPONSE_PAYLOAD, "Response had content-length"),
+                             new TestCase(CloseTime.DURING_RESPONSE_PAYLOAD, "Response had content-length"));
+    }
+
+    @Before
+    public void setup() throws Exception {
+        server = new Server();
+        server.init(testCase.closeTime);
+
+        netty = NettyNioAsyncHttpClient.builder()
+                                       .readTimeout(Duration.ofMillis(500))
+                                       .eventLoopGroup(SdkEventLoopGroup.builder().numberOfThreads(3).build())
+                                       .protocol(Protocol.HTTP1_1)
+                                       .buildWithDefaults(AttributeMap.builder().put(TRUST_ALL_CERTIFICATES, true).build());
+    }
+
+    @After
+    public void teardown() throws InterruptedException {
+        if (server != null) {
+            server.shutdown();
+        }
+        server = null;
+
+        if (netty != null) {
+            netty.close();
+        }
+        netty = null;
+    }
+
+    @Test
+    public void closeTimeHasCorrectMessage() {
+        server.closeTime = testCase.closeTime;
+        assertThat(captureException()).hasMessageContaining(testCase.errorMessage);
```

Comment:
Nit: Consider `assertThatThrownBy`.

Reply:
I didn't want the assertions running against the Error throwables that could be raised by the logic in captureException.